app/models/schedule.rb

```diff
@@ -137,4 +137,18 @@ def clean_up_task_jobs
     end
   end
 
+  def timeline
+    timeline = []
+    self.root_task.self_and_descendants.each do |task|
```

Comment:
You could probably clean this method up a bit if you used #collect instead of #each - then you wouldn't need to initialize timeline or manually return it (just return the result from the #collect block).